src/zabapgit_requirements.prog.abap

```diff
@@ -0,0 +1,205 @@
+
+      READ TABLE lt_installed WITH KEY component = <ls_requirement>-component
+                              ASSIGNING <ls_installed_comp>.
+      IF sy-subrc = 0.
+        <ls_status>-installed_release = <ls_installed_comp>-release.
+        <ls_status>-installed_patch = <ls_installed_comp>-extrelease.
+        <ls_status>-description = <ls_installed_comp>-desc_text.
+
+        <ls_status>-met = version_greater_or_equal( <ls_status> ).
+      ENDIF.
+
+      UNASSIGN <ls_installed_comp>.
+    ENDLOOP.
+  ENDMETHOD.
+
+  METHOD version_greater_or_equal.
+    DATA: lv_number TYPE numc4.
+
+    TRY.
+        MOVE EXACT: is_status-installed_release TO lv_number,
+                    is_status-installed_patch   TO lv_number,
+                    is_status-required_release  TO lv_number,
+                    is_status-required_patch    TO lv_number.
+      CATCH cx_sy_conversion_error.
+        " Cannot compare by number, assume requirement not fullfilled (user can force install
+        " anyways if this was an error)
+        rv_true = abap_false.
+        RETURN.
+    ENDTRY.
+
+    " Check if versions are comparable by number
+    IF is_status-installed_release > is_status-required_release
+       OR ( is_status-installed_release = is_status-required_release
+            AND ( is_status-required_patch IS INITIAL OR
+                  is_status-installed_patch >= is_status-required_patch ) ).
+
+      rv_true = abap_true.
+    ENDIF.
+  ENDMETHOD.
+
+  METHOD show_requirement_popup.
+    TYPES: BEGIN OF lty_color_line,
+             color TYPE lvc_t_scol.
+        INCLUDE TYPE gty_status.
+    TYPES: END OF lty_color_line,
+    lty_color_tab TYPE STANDARD TABLE OF lty_color_line WITH DEFAULT KEY.
+    DATA: lo_alv         TYPE REF TO cl_salv_table,
+          lo_column      TYPE REF TO cl_salv_column,
+          lt_color_table TYPE lty_color_tab,
+          lt_color_neg   TYPE lvc_t_scol,
+          lt_color_pos   TYPE lvc_t_scol,
+          ls_color       TYPE lvc_s_scol,
+          lx_ex          TYPE REF TO cx_root.
+    FIELD-SYMBOLS: <ls_line> TYPE lty_color_line.
+
+    ls_color-color-col = col_negative.
+    APPEND ls_color TO lt_color_neg.
+
+    ls_color-color-col = col_positive.
+    APPEND ls_color TO lt_color_pos.
+
+    CLEAR ls_color.
+
+    MOVE-CORRESPONDING it_requirements TO lt_color_table.
+
+    LOOP AT lt_color_table ASSIGNING <ls_line>.
+      IF <ls_line>-met = abap_false.
+        <ls_line>-color = lt_color_neg.
+      ELSE.
+        <ls_line>-color = lt_color_pos.
+      ENDIF.
+    ENDLOOP.
+    UNASSIGN <ls_line>.
+
+    TRY.
+        cl_salv_table=>factory(
+          IMPORTING
+            r_salv_table   = lo_alv
+          CHANGING
+            t_table        = lt_color_table
+        ).
+
+        lo_alv->get_columns(:
+          )->get_column( 'MET' )->set_short_text( 'Met' ),
+          )->set_color_column( 'COLOR' ),
+          )->set_optimize( ),
+        ).
```

Comment:
"Use chained statements mainly for declarations", http://help-legacy.sap.com/abapdocu_740/en/abenchained_statements_guidl.htm

this is quite confusing, also the last statement in the chain does not do anything? just `lo_alv->get_columns( )`